packages/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -76,6 +77,9 @@ class GoogleMap extends StatefulWidget {
   /// True if the map view should respond to scroll gestures.
   final bool scrollGesturesEnabled;
 
+  /// True if the map view should show zoom controls.
+  final bool zoomControlsEnabled;
```

Comment:
Since iOS doesn't support this. Let's add an assert so setting this property on iOS will throw in debug mode.

Reply:
From my point of view we have two choices to add debug assertions:
 - Assert during construction, which would in effect make the class not `const` anymore (Platform.isIOS is not const)
 - Create a separate property `_zoomControlsEnabled` like so: https://stackoverflow.com/a/27691564 and assert in the setter

Perhaps we're OK with a documentation update as done now instead?

Reply:
I like the second choice (`assert in the setter`).

Reply:
Can we implement a separate setter and assert there?